src/parse/rules/sh_rules.build_defs

```diff
@@ -55,21 +55,29 @@ def sh_binary(name, main, deps=None, visibility=None):
       main (str): The script to execute after all files have been uncompressed
       deps (list): Dependencies of this rule
       visibility (list): Visibility declaration of the rule.
+      shell (str): The shell to use when executing the produced script
     """
     tmpl_vars = {
-        'tarball_boundary': _TARBALL_BOUNDARY
+        'tarball_boundary': _TARBALL_BOUNDARY,
     }
 
-    cmds = ' && '.join([
-        # Inject bash code to untar the compressed files.
-        'echo "%s" > $OUT' % (_SH_BINARY_TMPL_PREFIX % tmpl_vars),
-        # Inject the user defined script.
-        'cat $SRCS >> $OUT',
-        # Inject the tar ball boundary.
-        'echo "%s" >> $OUT' % (_SH_BINARY_TMPL_SUFFIX % tmpl_vars),
-        # Compress the dependent files and dump out into the bash script.
-        'find . -type f | grep -v $SRCS | grep -v $(echo $OUT | sed "s|$TMP_DIR||g") | sed "s/^\.\///g" | tar zc -f - --files-from - >> $OUT',
-    ])
+    if deps:
+        cmds = ' && '.join([
+            # Inject bash code to untar the compressed files.
+            'echo "%s" > $OUT' % (_SH_EXEC % shell),
+            'echo "%s" > $OUT' % (_SH_BINARY_TMPL_PREFIX % tmpl_vars),
+            # Inject the user defined script.
+            'cat $SRCS >> $OUT',
+            # Inject the tar ball boundary.
+            'echo "%s" >> $OUT' % (_SH_BINARY_TMPL_SUFFIX % tmpl_vars),
+            # Compress the dependent files and dump out into the bash script.
+            'find . -type f | grep -v $SRCS | grep -v $(echo $OUT | sed "s|$TMP_DIR||g") | sed "s/^\.\///g" | tar zc -f - --files-from - >> $OUT',
+        ])
+    else:
+        cmds = ' && '.join([
```

Comment:
This branch is short enough that it might be easier to do it as one line. Up to you.